docs/sql-connector.md

```diff
@@ -0,0 +1,662 @@
+# Build a connector for relational databases
+
+This tutorial walks you through the MySQL connector implementation to teach you
+how to develop a connector for relational databases.
+
+## Understand a connector's responsibilities
+
+In LoopBack, models encapsulate business data and logic as JavaScript properties
+and methods. One of the powerful features of LoopBack is that application 
+developers don't have to implement all behaviors for their models as a lot of 
+them are already provided by the framework out of box with data sources and 
+connectors. For example, a model automatically receives the create, retrieve, 
+update, and delete (CRUD) functions if it is attached to a data source for a
+database. LoopBack abstracts the persistence layer and other backend services, 
+such as REST APIs, SOAP web services, and storage services, and so on, as data 
+sources, which are configurations of backend connectivity and integration. 
+Each data source is backed a connector which implements the interactions between
+Node.js and its underlying backend system. Connectors are responsible for 
+mapping model method invocations to backend functions, such as database 
+operations or call to REST or SOAP APIs. The following diagram illustrates how
+connectors fit into the big picture of LoopBack API framework.
+ 
+![connector-architecture](connector-architecture.png)
```

Comment:
This should be a separate document that we link to. Keep this on topic (building a SQLConnector).